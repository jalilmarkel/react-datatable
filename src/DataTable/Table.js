import styled from 'styled-components';

const TableStyle = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  ${props => props.disabled && 'pointer-events: none'};
  ${props => props.disabled && 'opacity: 0.4'};
`;

export default TableStyle;
