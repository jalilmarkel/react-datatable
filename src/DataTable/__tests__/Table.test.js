import 'jest-styled-components';
import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Table from '../Table';

afterEach(cleanup);

test('component <Table /> should render correctly', () => {
  const { container } = render(<Table />);
  expect(container.firstChild).toMatchSnapshot();
});

test('component <Table disabled={true} /> should render correctly ', () => {
  const { container } = render(<Table disabled />);
  expect(container.firstChild).toMatchSnapshot();
});
