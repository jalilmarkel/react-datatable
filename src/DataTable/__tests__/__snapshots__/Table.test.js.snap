// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Table /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
}

<div
  class="c0"
/>
`;

exports[`component <Table disabled={true} /> should render correctly  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  pointer-events: none;
  opacity: 0.4;
}

<div
  class="c0"
  disabled=""
/>
`;
